includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2220,6 +2220,14 @@ public function get_level3_data_from_order( WC_Order $order ): array {
 			$tax_amount      = WC_Payments_Utils::prepare_amount( $item->get_total_tax(), $currency );
 			$discount_amount = WC_Payments_Utils::prepare_amount( $subtotal - $item->get_total(), $currency );
 
+			// This is a special case, to prevent Stripe error when $unit_cost is negative. The unit cost can be
+			// negative if the item has a negative price.
+			if ( 0 > $subtotal ) {
+				// To prevent the Stripe API error, set the absolute value of the unit cost as discount.
+				$discount_amount = abs( $unit_cost );
+				$unit_cost       = 0;
```

Comment:
https://stripe.com/docs/level3 says unit cost shall be greater than zero, so this approach will not work. Can you test this please before we proceed with reviews?

Reply:
Good point. I tested it and it allowed me to create the payment_intent including the level3 data: ["pi_3KxXtsQsDOQXPzI10nBXFZqc"](https://dashboard.stripe.com/test/payments/pi_3KxXtsQsDOQXPzI10nBXFZqc). This is an excerpt of the payment_intent response from the Stripe logs:

````
 "level3": {
          "customer_reference": "190",
          "line_items": [
            {
              "discount_amount": 2000,
              "product_code": "179",
              "product_description": "You Saved 5% OFF (Discount",
              "quantity": 1,
              "tax_amount": 0,
              "unit_cost": 0
            },
            {
              "discount_amount": 0,
              "product_code": "17",
              "product_description": "Sunglasses",
              "quantity": 1,
              "tax_amount": 0,
              "unit_cost": 9000
            }
          ],
          "merchant_reference": "190",
          "shipping_address_zip": "90210",
          "shipping_amount": 100,
          "shipping_from_zip": "94110"
        },
````
It seems the documentation is not up to date or accurate?

Reply:
THanks for testing, probably the documentation is not up to date. I'll ping stripe for updating it.